tests/raftstore/cluster.rs

```diff
@@ -297,7 +298,8 @@ impl<T: Simulator> Cluster<T> {
 
         let err = resp.get_header().get_error();
         if !err.has_not_leader() {
-            return false;
+            self.reset_leader_of_region(region_id);
```

Comment:
reset the cached leader in Cluster's, so next time query goes to Pd, chance that we'll get the right leader.